src/refdb_fs.c

```diff
@@ -866,12 +875,20 @@ static int refdb_fs_backend__unlock(git_refdb_backend *backend, void *payload, i
 	git_filebuf *lock = (git_filebuf *) payload;
 	int error = 0;
 
-	if (success == 2)
-		error = refdb_fs_backend__delete_tail(backend, lock, ref->name, NULL, NULL);
-	else if (success)
-		error = refdb_fs_backend__write_tail(backend, ref, lock, update_reflog, NULL, NULL, sig, message);
-	else
-		git_filebuf_cleanup(lock);
+	git__refdb_flags flags = (GIT_REFDB__SKIP_LOCK | GIT_REFDB__FORCE_WRITE);
+	if (!update_reflog) {
+		flags |= GIT_REFDB__SKIP_REFLOG;
+	}
+	if (success == 2) {
+		error = refdb_fs_backend__delete_tail(lock, backend, ref->name, flags, NULL, NULL);
+	} else if (success) {
+		error = refdb_fs_backend__write_tail(lock, backend, ref, flags, NULL, NULL, sig, message);
+		if (error == 0) {
+			error = loose_commit(lock, ref);
```

Comment:
Why exactly do we need this now and why wasn't it here before? I honestly don't quite know, and `loose_commit` is a rather weird function name that doesn't help

Edit: oh, it makes a bit more sense when one notices that things like `loose_lock` etc. also exist. I still am a bit puzzled, mainly because I don't know this code